va/va_prot.h

```diff
@@ -0,0 +1,353 @@
+} VAProtectedSessionExecutecBuffer;
+
+/** \brief Buffer for vaProtectedSessionHwUpdate() */
+typedef struct _VAProtectedSessionHwUpdateBuffer {
+    /** \brief Input buffer of VAProtectedSessionBufferIn structure */
+    VAProtectedSessionBuffer    input;
+    /** \brief Output buffer of VAProtectedSessionBufferOut structure */
+    VAProtectedSessionBuffer    output;
+    /** \brief Return the result of this function. It is IHV's implementation specific */
+    uint32_t                    status;
+    uint32_t                    va_reserved[VA_PADDING_LOW];
+} VAProtectedSessionHwUpdateBuffer;
+
+/**
+ * \brief Create a protected session
+ *
+ * Create a protected session
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] config_id             configuration for the protected session
+ * @param[out] protected_session    created protected session id upon return
+ */
+VAStatus vaCreateProtectedSession(
+    VADisplay dpy,
+    VAConfigID config_id,
+    VAProtectedSessionID *protected_session
+);
+
+/**
+ * \brief Destroy a protected session
+ *
+ * Destroy a protected session
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] protected_session     protected session to be destroyed
+ */
+VAStatus vaDestroyProtectedSession(
+    VADisplay dpy,
+    VAProtectedSessionID protected_session
+);
+
+/**
+ * \brief Attach a protected session to the display
+ *
+ * Attach a protected session to the display to enable protection
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] reserved              reserved for future usage
+ * @param[in] protected_session     the protected session to attach
+ */
+VAStatus vaAttachProtectedSession(
+    VADisplay dpy,
+    VAContextID reserved,
+    VAProtectedSessionID protected_session
+);
+
+/**
+ * \brief Detach the protected session of the display
+ *
+ * Detach protected session of the display to disable protection
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] reserved              reserved for future usage
+ */
+VAStatus vaDetachProtectedSession(
+    VADisplay dpy,
+    VAContextID reserved
+);
+
+/**
+ * \brief TEE execution for the particular protected session
+ *
+ * TEE execution for the particular protected session.
+ * This is a synchronous API.
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] protected_session     the protected session
+ * @param[in,out] buf_id            the VA buffer
+ */
+VAStatus vaProtectedSessionExecute(
+    VADisplay dpy,
+    VAProtectedSessionID protected_session,
+    VABufferID buf_id
+);
+
+/**
+ * \brief Update protected hardware settings
+ *
+ * Allows the driver to update IHV specific hardware status for
+ * the protected session. This is a synchronous API.
+ *
+ * @param[in] dpy                   the VA display
+ * @param[in] protected_session     the protected session
+ * @param[in,out] buf_id            the VA buffer
+ */
+VAStatus vaProtectedSessionHwUpdate(
```

Comment:
I am still not convinced that separating exec and hwupdate is a right thing to do. Let's see comments when these functions should actually be called... maybe this will help to make a decision...

Reply:
HwUpdate related definitions are removed.